### ISSUE TITLE
Dates are being converted to {} after mutation

### ISSUE THREAD

**Issue by garrettg123** [klis87/normy#7]
Here's my config:
```tsx
    <QueryNormalizerProvider
      queryClient={queryClient}
      normalizerConfig={{
        normalize: typeof window !== 'undefined',
        devLogging: true,
        getNormalizationObjectKey: (obj: { id?: string; createdAt?: Date }) => {
          const key =
            typeof obj.id === 'string' && obj.createdAt instanceof Date
              ? obj.id
              : undefined

          if (!key) {
            return
          }

          console.log('Normalizing %o', obj)
          return key
        },
      }}>
```
Which prints my log with the date, and then the dev log, which shows the date being gone.
```
Normalizing 
{id: 'cloeog1mm0005ckahzxq197b1', createdAt: Tue Oct 31 2023 14:43:27 GMT-0400 (Eastern Daylight Time), updatedAt: Tue Oct 31 2023 16:43:02 GMT-0400 (Eastern Daylight Time), …}

set query: [["task","get"],{"input":"cloeog1mm0005ckahzxq197b1","type":"query"}] 
with data: 
{id: 'cloeog1mm0005ckahzxq197b1', createdAt: {…}, updatedAt: {…}, type: 'REMINDER', title: 'LOI coming soon', …}
actions
: 
[]
assignedDate
: 
{}
completedDate
: 
{}
createdAt
: 
{}
```

**Comment by klis87** [comment]
The first question is, how `Date` instance is present in your data? Generally this lib will work only for JSON serializable data, things like `Date` could not work

**Comment by klis87** [comment]
hmm, but you need to get `Date` from backend, is not that stringified date?

**Comment by klis87** [comment]
Ok I get it. I would like to know now, if the issue is that `normy` does not work with `date` at all at the moment, or there is a problem with:
```js
getNormalizationObjectKey: (obj: { id?: string; createdAt?: Date }) => {
          const key =
            typeof obj.id === 'string' && obj.createdAt instanceof Date
              ? obj.id
              : undefined

          if (!key) {
            return
          }

          console.log('Normalizing %o', obj)
          return key
        },
```

I guess you want to normalize objects only with `createdAt`. Can you remove it for now and let me know if this does not work as well? If yes, let me know please if issue is already after queries are fetched, or only after a mutation. Then I will try to fix it, as I think that dates should be supported, I cannot see why not.

**Comment by klis87** [comment]
Ok thanks, I will debug it myself and try to fix, or... implement it? I am just not sure this is a big or feature :D as I did not intend to work with dates.

**Comment by klis87** [comment]
> No problem! Any chance it will work with 0.10 instead of 0.11? Because TRPC is only on react-query v4.

Yes, I will backport it as well as 0.10.1 . I think I will support both versions for now, as many people are still on react-query 4

**Comment by klis87** [comment]
Fixed in https://github.com/klis87/normy/releases/tag/%40normy%2Freact-query%400.11.2 , please let me know if it works now. I fixed it both for 0.10 and 0.11 versions.
